src/ol/source/DataTile.js

```diff
@@ -239,6 +239,7 @@ class DataTileSource extends TileSource {
         gutter: this.getGutterForProjection(sourceProj),
         getTileFunction: (z, x, y, pixelRatio) =>
           this.getTile(z, x, y, pixelRatio, sourceProj),
+        clipExtent: !this.getWrapX(),
```

Comment:
You tried to explain it here and in #15417, but I'm still failing to understand what `wrapX` has to do with this.

Don't we always want to clip to the source extent? Why make it optional?

Reply:
As described in #15424 I do not think it `wrapX` should have anything to do with this, that is a misuse of `wrapX` and it should be a separate option.

We do not clip (and never have) to the tile grid extend in the canvas tile renderer, the tile grid extent merely prevents tiles which are completely outside the extent being requested, it does not stop tiles partly in the extent being rendered in full.  Typical example would be a regional (at detail level) map using a standard EPSG:3857 tile grid origin.  But as in the example there is a requirement for that with GeoTIFF sources (and it should not depend on `wrapX`).